libs/blocks/aside/aside.js

```diff
@@ -105,11 +111,56 @@ function decorateBlockBg(block, node) {
   }
 }
 
+function addPromobar(sourceEl, parent) {
+  const newPromo = sourceEl.cloneNode(true);
+  parent.appendChild(newPromo);
+}
+
+function checkViewportPromobar(foreground) {
+  const childCount = foreground.childElementCount;
+  const { children } = foreground;
+  if (childCount < 2) addPromobar(children[childCount - 1], foreground);
+  if (childCount < 3) addPromobar(children[childCount - 1], foreground);
+}
+
+function combineTextBocks(iconArea, textBlocks, mediaPort) {
+  const contentArea = createTag('p', { class: 'content-area' });
+  const textArea = createTag('p', { class: 'text-area' });
+  textBlocks[0].parentElement.prepend(contentArea);
+  textBlocks.forEach((textBlock) => {
+    textArea.appendChild(textBlock);
+    if (textBlock.nodeName === 'P') textBlock.classList.add(`body-${blockConfig.promobar[mediaPort][1]}`);
+    else textBlock.classList.add(`heading-${blockConfig.promobar[mediaPort][0]}`);
```

Comment:
This is not too readable, maybe:
```
if (textBlock.nodeName === 'P') {
    textBlock.classList.add(`body-${blockConfig.promobar[mediaPort][1]}`);
} else {
    textBlock.classList.add(`heading-${blockConfig.promobar[mediaPort][0]}`);
}
```
Also, as far as I can tell this is the only place where the blockConfig is used. Do we need to add the the config object for this. It looks like when this is called, the code loops through the children, and then accesses the mediaport const above (looks like there is 3 expected?) and then passes that media port to this function. 

I guess what I am getting at is there is a lot of "known" quantities here but the mediaPort parameter and accessing it by array index feels like it is trying to solve for something more dynamic.

Reply:
Made the suggested changes to improve redability. I had added the config to block config as it contains the styling of all aside formats. Since this style was specific to this function of promobar I have moved it inside combineTextBocks function.